src/platform/julius.c

```diff
@@ -405,6 +406,21 @@ static int pre_init(const char *custom_data_dir)
         return game_pre_init();
     }
 
+
+    char *prev = NULL;
+    prev = getcwd(prev, 0);
+
+    // AppImage support
+    if (chdir("../share/julius") == 0) {
```

Comment:
This should be relative to the executable's location, not the current working directory.

Reply:
I doubt we will ever have the game data shipped with the AppImage because it's copyrighted. And if we provide alternative graphics, I think whoever holds the rights to C3 nowadays will come after us because we then basically provide "their" game for free. Right now that's not the case because Julius requires the original game files.

Besides that: the AppImage is run from... where? `/usr/bin`? `/usr/local/bin`? Those are not locations where your average user has write access, and we need that for the savegames, user-created maps and settings files.

Reply:
Even though we won't ship the game data, users can add the game data to the AppImage *after the fact*. Especially useful in the future in combination with the in-place updates that AppImage supports.

They run from a SquashFS mounted by FUSE. Even if FUSE is not available on your OS (it's installed by default on most distributions), you can anyway run the AppImage using the `--appimage-extract-and-execute`. The mount point is usually something like `/dev/loopX`. See [the doc](https://docs.appimage.org/user-guide/run-appimages.html) for more information.

The data is stored in `$HOME` by default. This can be changed and AppImage's [can be run portably](https://docs.appimage.org/user-guide/run-appimages.html) if necessary.

Reply:
> Even though we won't ship the game data, users can add the game data to the AppImage _after the fact._
 
Can you explain how? I toyed around with the appimage: it's basically running in /usr/bin from a sandboxed Linux system, which is read-only. It doesn't make sense to try `../share/julius` as data directory because we need write access.

Reply:
> it's basically running in /usr/bin 

This sounds impossible to me. Are you sure it was not running from a mounted location?

> Can you explain how?

Yes, just extract the content with `--appimage-extract`, add the game data, then repack it using `./linuxdeploy-x86_64.AppImage --appdir AppDir --output appimage`.

There might be a way to directly add data to an existing AppImage but I don't have enough experience with this tool so I don't know how one would do this.

Reply:
I did some tests:

- All data in the virtual file system of the app image is *read-only*. If you pack the whole C3 directory with data files etc and use that, then you won't be able to save the game, save settings, save created scenarios, etc. Basically useless.
- The working directory of an appimage executable works exactly the same as for a normal executable: when started from the command-line, it's the directory where you run the command from. When started from a file manager: the directory of the appimage executable. "../share/julius" is _never_ going to resolve to a path inside the appimage file system.

Therefore: please remove this check for "../share/julius".